File: Core-backend-layer/Programming-fundamentals/Python/OOP/src/classes_and_objects/product.py
class Product:
    def __init__(self, name, price, product_id, manufacturer, production_date):
        self.name = name
        self.price = price
        self.product_id = product_id
        self.manufacturer = manufacturer
        self.production_date = production_date

    def display_info(self):
        print(f"Product ID: {self.product_id}, Product: {self.name}, Price: ${self.price}, "
              f"Manufacturer: {self.manufacturer if self.manufacturer is not None else 'N/A'}, "
              f"Production date: {self.production_date if self.production_date is not None else 'N/A'}")

File: Core-backend-layer/Programming-fundamentals/Java/OOP/src/ClassesAndObjects/Product.java
import java.time.LocalDateTime;

public class Product {
    // Fields
    String name;
    double price;
    final int productId; // Using for a constant field
    String manufacturer; // Nullable field
    LocalDateTime productionDate;

    // Constructor
    public Product(String name, double price, int productId, String manufacturer, LocalDateTime productionDate) {
        this.name = name;
        this.price = price;
        this.productId = productId;
        this.manufacturer = manufacturer;
        this.productionDate = productionDate;
    }

    // Method
    public void displayInfo() {
        System.out.println("Product ID: " + productId + ", Product: " + name + ", Price: $" + price
                + ", Manufacturer: " + (manufacturer != null ? manufacturer : "N/A")
                + ", Production date: " + (productionDate != null ? productionDate : "N/A"));
    }
}